Sources/Brave/Frontend/Browser/Search/Voice Search/SpeechRecognizer.swift

```diff
@@ -0,0 +1,300 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import Speech
+import SwiftUI
+import os.log
+
+protocol SpeechRecognizerDelegate: AnyObject {
+  func speechRecognizerDidFinishQuery(query: String)
+}
+
+class SpeechRecognizer: ObservableObject {
+  
+  private let log = Logger(subsystem: Bundle.main.bundleIdentifier!, category: "voiceRecognizer")
+  
+  enum RecognizerError: Error {
+    case failedSetup
+    case microphoneAccessDenied
+    case recognizerIsUnavailable
```

Comment:
Do we ever behave different depending on error code? If not I do not think we need it, we can just print any error we want at the time we have to throws it so it could be removed

Reply:
I wanted to use this enum for the case where online speech recognition error case. Thanks for noticing. 

Instead of removing the enum I added functionality to enable permission for this case

Fix at https://github.com/brave/brave-ios/pull/7551/commits/9919db2ea0f5f45e36ab3fba198c73aba96ac31a